File: 5. String/palindromeSubstrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countSubstrings = function(s) {
    let ret = 0;
    for (let i = 0; i < s.length; i++) {
        for (let j = 0; j < 2; j++) {
            let left = i;
            let right = i + j;
            while (left >= 0 && right < s.length && s[left] === s[right]) {
                ret++;
                left--;
                right++;
            }
        }
    }
    return ret;
};

File: 5. String/palindrome_substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        ret = 0
        for i, c in enumerate(s):
            for j in range(2):
                right = i + j
                left = i
                while left >= 0 and right < len(s) and s[right] == s[left]:
                    ret += 1
                    left -= 1
                    right += 1

        return ret
